Simple-PHP-IPAM/upgrade.sh

```
@@ -237,10 +225,77 @@ if [[ -n "$CHOWN_BIN" ]]; then
+    if [[ "$driver_choice" == "m" || "$driver_choice" == "p" ]]; then
+      target_driver=$([[ "$driver_choice" == "m" ]] && echo "mysql" || echo "pgsql")
+      echo ""
+      read -r -p "Target DSN (e.g. mysql:host=127.0.0.1;dbname=ipam): " target_dsn </dev/tty
+      read -r -p "Target username: " target_user </dev/tty
+      read -r -s -p "Target password: " target_pass </dev/tty
+      echo ""
+
+      if [[ -z "$target_dsn" ]]; then
+        echo "No DSN provided. Skipping driver migration."
+      else
+        local src_dsn
```

Shellcheck (0.11.0)

[error] 'local' is only valid in functions.

(SC2168)